### ISSUE TITLE
Chapter 3.8.2: DB Choice - transactions

### ISSUE THREAD

**Issue by bbros-dev** [LukeMathWalker/zero-to-production#66]
If you were to select one of the more modern/recent DB's many of your users would still go with a relational DB. 

Simply because they require transactions - or more specifically have a product owners/clients who insist on having a transactions capable store.  And no-one has enough years ahead of them to sit in endless meetings conjecturing about product blue-sky projections and contingencies - so transactional DB it is.

Perhaps a few lines about transactions and async API's?

**Comment by bbros-dev** [comment]
One could argue that this is resolved when issue #65 is closed - depending on the footnote content.

However, given the ubiquity of DB transactions it would still seem strange not to have a section dedicated to some discussion (no matter how brief), of their use in async contexts, and the rust ecosystem crates supporting that use case.

**Comment by LukeMathWalker** [comment]
This will be addressed in the next release (going out tomorrow).  
I am going to add a mention to transactions in Chapter 3 (pretty much for the golden reason you mentioned - `no-one has enough years ahead of them to sit in endless meetings conjecturing about product blue-sky projections and contingencies - so transactional DB it is.`) but we will discuss them at the proper level of detail at the end of Chapter 7.

I understand that, as an experienced dev going through the book, you might sometimes wonder "Why is not talking about X here?". 
The general principle I have tried to follow throughout the book is "do it wrong first, then show me what I need to do it right". I feel it makes the material easier to contextualise - last-minute explanations seem to stick longer with readers (it'd be nice to have data to back it up tough).
We never make more than one state-mutating query in Chapter 3, so it's difficult to appreciate what transactions provide there. In Chapter 7, with confirmation emails and a more complex flow, we have a better opportunity to discuss what transactions bring to the table.

Chapter 7 provides us as well with a good opportunity to discuss some of the topics we touched on in #65 - keen to get your feedback there.
